com.reprezen.swagedit/src/com/reprezen/swagedit/validation/Validator.java

```diff
@@ -188,9 +188,14 @@ protected boolean hasArrayType(AbstractNode node) {
     protected void checkObjectTypeDefinition(Set<SwaggerError> errors, AbstractNode node) {
         if (node instanceof ObjectNode) {
             JsonPointer ptr = node.getPointer();
-            if (ptr != null && ptr.toString().startsWith("/definitions")) {
-                checkMissingType(errors, node);
-                checkMissingRequiredProperties(errors, node);
+            if (ptr != null) {
+                if (ptr.toString().startsWith("/definitions")) {
+                    checkMissingType(errors, node);
+                    checkMissingRequiredProperties(errors, node);
+                } else if (ptr.toString().endsWith("/schema")) {
```

Comment:
I suggest extracting both checks (`ptrStr.startsWith("/definitions")` and `ptrStr.endsWith("/schema")`) into `isSchemaDefinition()`